### PR TITLE
[formatter]: fixing crash when upstreamClusterInfo is null

### PR DESCRIPTION
**Pull request by tehnerd**
Signed-off-by: Nikita V. Shirokov <tehnerd@tehnerd.com>

This PR fixing crash, but i've contacted security email list already, and decision was that it is ok to fix this in public PR.

Commit Message:
In our environment we bumped in situation when optional upstreamCluster has a value, but value is equal to nullptr.
this regression was introduced in pretty recent commit: https://github.com/envoyproxy/envoy/commit/980c84dca7305e1dd29afe2ba9ce0d2d4a36bc9f#diff-888dc1d9b82ce89abbe565c8c52a11dd16120d2d23b794e1a8c46e5ac0bb039a where such case was not properly tested.

In this diff i'm fixing the issue by explicitly testing cluster against nullptr (and it looks like this is common case for other usage of upstreamClusterInfo() in formatter)

Additional Description:

Risk Level:
LOW
Testing:
UT
Docs Changes:
Release Notes:
Platform Specific Features:

### CODE REVIEW

**Comment by tehnerd** [comment]
cc: @alyssawilk  @cpakulski

**Review by cpakulski — APPROVED** [review]
LGTM.

**Review by alyssawilk — APPROVED** [review]

**Comment by alyssawilk** [comment]
As said "offline" nice catch!  Thanks for the fix and test.
Heading out (EST) but I'll merge tomorrow if none of the other @envoyproxy/maintainers beat me to it :-)

File: source/common/formatter/substitution_formatter.cc
Comment (antoniovicente) [inline source/common/formatter/substitution_formatter.cc:756]: It seems unfortunate that we can have upstreamClusterInfo be set to nullptr, but this behavior is documented in upstreamClusterInfo's interface definition.

**Review by antoniovicente — APPROVED** [review]
Thanks for the fix.

**Comment by alyssawilk** [comment]
Coverage has failed twice, can you try a main merge?  Or @junr03 is there a known issue here?

File: source/common/formatter/substitution_formatter.cc
Comment (ggreenway) [reply source/common/formatter/substitution_formatter.cc:756]: That's a very unintuitive interface: an optional pointer that can be null. Can this be cleaned up to be not-optional, or not-nullable?

**Pull request merged**
